doc/en/fixture.rst

```diff
@@ -216,20 +216,189 @@ functions take the role of the *injector* and test functions are the
 .. _`conftest.py`:
 .. _`conftest`:
 
-``conftest.py``: sharing fixture functions
-------------------------------------------
+Fixture availabiility
+---------------------
 
-If during implementing your tests you realize that you
-want to use a fixture function from multiple test files you can move it
-to a ``conftest.py`` file.
-You don't need to import the fixture you want to use in a test, it
-automatically gets discovered by pytest. The discovery of
-fixture functions starts at test classes, then test modules, then
-``conftest.py`` files and finally builtin and third party plugins.
+Fixture availability is determined from the perspective of the test. A fixture
+is only available for tests to request if they are in the scope that fixture is
+defined in. If a fixture is defined inside a class, it can only be requested by
+tests inside that class. But if a fixture is defined inside the global scope of
+the module, than every test in that module, even if it's defined inside a class,
+can request it.
+
+Similarly, a test can also only be affected by an autouse fixture if that test
```

Comment:
The concept of autouse fixtures has a few nuances, and the implications (both positive and potentially negative) of it may not be immediately apparent. I wanted to explicitly mention this here and provide a link to the order section for them, as it touches on this topic in more details

I don't think it not being mentioned yet is a problem necessarily, but perhaps the autouse fixture section should be moved above this section?

Reply:
I think moving autouse above this would solve it. 👍